Makefile

```
@@ -169,10 +169,14 @@ cfn/lint: | guard/program/cfn-lint
 
 ## Runs eclint against the project
 eclint/lint: | guard/program/eclint guard/program/git
-eclint/lint: ECLINT_PREFIX ?= git ls-files -z | xargs -0
+eclint/lint: HAS_UNTRACKED_CHANGES ?= $(shell git diff-index --quiet HEAD -- || echo "true")
+eclint/lint: ECLINT_PREFIX ?= git ls-files -z
 eclint/lint:
 	@ echo "[$@]: Running eclint..."
-	cd $(PROJECT_ROOT) && $(ECLINT_PREFIX) eclint check
+	cd $(PROJECT_ROOT) && \
+	[ -z "$(HAS_UNTRACKED_CHANGES)" ] && \
```

May want to reverse this logic also, so an actual eclint failure doesn't just output `untracked changes detected!`

```suggestion
	[ -z "$(HAS_UNTRACKED_CHANGES)" ] || (echo "untracked changes detected!" && exit 1)
```

---

💡I was fighting with this for a while trying to get the logic all on one line.. it just clicked that I don't need to b/c make will error out before it gets to the next step. frustrating that subshell exiting doesn't work the way I wanted. 

learned the hard way that this will always print both `foo set` _and_ `foo not set`: 
```bash
foo="bar"

[ -n "$foo" ] && (echo "foo set" && exit 1) || (echo "foo not set")
```

---

The subshell _is_ doing the right thing and exiting nonzero. The second echo is triggered by the nonzero exit and ||. You can flip the || to &&, and it will not print when foo is set, but then when foo is unset nothing will print....

As shellcheck warns, it is chaining `&&` / `||` that does not do what you think it does... Best to avoid it wherever possible! Especially here, Let make do its make thing.